paper/paper.md

```diff
@@ -0,0 +1,204 @@
+
+The library is actively used to modernize the Fortran-based Gysela plasma simulation code (TODO: add a citation). This simulation code relies heavily on high-dimensional arrays that span multiple representations, including Fourier, spline, Cartesian, and various curvilinear meshes. In the legacy Fortran implementation, these arrays were manipulated using implicit conventions, making it difficult to enforce correctness at the API level. DDC enables a more explicit, strongly-typed representation of these arrays, ensuring at compile-time that function calls respect the expected dimensions. This reduces indexing errors and improves code maintainability, particularly in large-scale scientific software.
+
+## DDC Core key features
+
+The DDC library is a C++ library designed for expressive and safe handling of multidimensional data. Its core component provides flexible data containers along with algorithms built on top of the performance portable Kokkos library. The library is fully compatible with Kokkos and does not attempt to hide it, allowing users to leverage Kokkos' full capabilities while benefiting from DDC’s strongly-typed, labelled multidimensional arrays.
+
+### Containers
+
+DDC offers two multidimensional containers designed over the C++ 23 multidimensional array `std::mdspan`:
+
+- `Chunk` an owning container, i.e. it manages the lifetime of the underlying memory allocation,
+- `ChunkSpan` is a non-owning container view over existing memory allocation.
+
+### Strongly-typed labelled indices
+
+DDC employs strongly-typed multi-indices to label dimensions and access data. It introduces two types of indices to access the container's data:
+
+- `DiscreteVector` indices:
+  - strongly-typed labelled integers,
+  - provide a multidimensional array access semantics,
+  - always as fast access as raw multidimensional array,
+- `DiscreteElement` indices:
+  - strongly-typed labelled keys or opaque identifiers,
+  - provide an associative access semantics, as keys in a map container,
+  - potentially slower access, depending on the type of set of `DiscreteElement`.
+
+Unlike `DiscreteVector` indices, users cannot directly interpret the internal representation of `DiscreteElement` and must reason about them based solely on their relative position.
+
+### Sets of `DiscreteElement`
+
+The semantics of DDC containers is to associate data to a set of `DiscreteElement` indices. Let us note that the set of all possible `DiscreteElement` has a total order that is typically established once and for all at program initialisation. Thus to be able to construct a DDC container one must provide a multidimensional set of `DiscreteElement` indices, only these indices can be later used to access the container’s data.
+
+The set of `DiscreteElement` is a customization point of the library. It takes the form of a Cartesian product of the different dimensions. DDC predefines the following sets:
```

Comment:
I am not sure what is meant here

Reply:
What part ? The "customization point" ?

Reply:
Yes, the sentence "The set of `DiscreteElement` is a customization point of the library. "

Reply:
So DDC provides `DiscreteDomain`, `StridedDiscreteDomain` and `SparseDiscreteDomain` but a user can provide its own domain for some specific pattern.

Reply:
I guess you should clarify this. I'm not sure how much your typical reader will understand of the concept of "customization point" and what it means in this specific context.

Reply:
```suggestion

The library provides multiple ways to group `DiscreteElement`s into sets. Each set takes the form of a Cartesian product of the sets along each dimensions. DDC predefines the following sets:
```
?